react-cors-spa/src/App.js

```diff
@@ -22,38 +22,50 @@ import logoCF from './logoCloudFront.png';
 import './App.css';
 import axios from 'axios';
 import { useState, useEffect } from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
 
 // To be replaced by the endpoint of the API deployed through the CloudFormation Template
-const APIEndPoint = 'to be replaced with your api endpoint here'
+const APIEndPoint = 'https://api.quotable.io/random'
+
+// Create a client
+const queryClient = new QueryClient();
 
 function App() {
   return (
-    <div className="App">
-        <header className="App-header">
-          {APIEndPoint.startsWith('http') &&
-            <APIResult />
-          }
-          <img src={logo} className="App-MainLogo" alt="logo" />
-        </header>
-        <p>This react-based application is hosted in an S3 bucket exposed through a CloudFront distribution</p>
-        <div className="logos">
-            <img src={logoS3} className="App-logoR2L" alt="logo S3" />
-            <img src={logoCF} className="App-logoL2R" alt="logo CloudFront" />
-        </div>
-    </div>
+    <QueryClientProvider client={queryClient}>
+      <div className="App">
+          <header className="App-header">
+            {APIEndPoint.startsWith('http') &&
+              <APIResult />
+            }
+            <img src={logo} className="App-MainLogo" alt="logo" />
+          </header>
+          <p>This react-based application is hosted in an S3 bucket exposed through a CloudFront distribution</p>
+          <div className="logos">
+              <img src={logoS3} className="App-logoR2L" alt="logo S3" />
+              <img src={logoCF} className="App-logoL2R" alt="logo CloudFront" />
+          </div>
+      </div>
+      {/*Floating Mode running locally*/}
+      {process.env.NODE_ENV === 'development' && (
+      <ReactQueryDevtools initialIsOpen={false} position="bottom-right" />
+    )}
+  </QueryClientProvider>
   );
 }
 
 const APIResult = () => {
```

Comment:
Setup of React Query looks good but you didn't actually use it anywhere. See the first example on this page: https://tanstack.com/query/latest/docs/framework/react/quick-start

Reply:
You can ignore the useMutation, but the rest of the Todos component is basically set up to serve the same function as our APIResult component so you can really lean on that example. It gives you a way to display the data too.

Reply:
The axios call will be in place of the `getTodos` function. It may also be worth checking out the documentation for useQuery to see the other properties that the `query` object will have besides `.data`.

Reply:
Sorry about that - is it alright if the additional component is outside of the app component? Or should it be nested inside of the app component?

Reply:
Tbh it should probably be in a separate file. What you did by having them sequentially is okay in some cases, but these components are different enough they should probably go in separate files.